### PR TITLE
French Translation and Image Optimisations

### PR DESCRIPTION
**Pull request by PoorPocketsMcNewHold**
A more complete French translation, as requested.
Little notes however : - The Image optimizations was an automatic pull request generated by ImgBotApp, if you don't want them, you can just simply replace those files provided in the pull request by the original ones (Or, I think I can do some sort of revert commit).
- I didn't know where I was supposed to put the original extension config translation, so I've put it in the new fr folder, as fr.txt
- I don't know GitHub that much for really understanding the GitHub Wiki. Your original didn't fork when I've forked your project, so I assume it wouldn't have been included in the pull request if I've had made that pull request. The issue however, is that I don't know how to “add” new page to your wiki (And surely shouldn't be supposed to be able to anyway), so should i just add the French translation of the wiki, next to the original English translation?

### CODE REVIEW

**Comment by PoorPocketsMcNewHold** [comment]
Oh, and despite being asked in #36 due to the issue with the [web translator output](https://github.com/Lusito/web-ext-translator/issues/30), I didn't fix it in the fr.txt.

**Comment by SuperSandro2000** [comment]
You could have split this into two PRs.

**Comment by PoorPocketsMcNewHold** [comment]
> You could have split this into two PRs.

Indeed, I could. The thing was this was an automated pull request from the Bot. So, I would have to wait the first pull request to be applied, before updating my fork with the Bot Pull request, and then create a second pull request for that part.

**Comment by SuperSandro2000** [comment]
> > You could have split this into two PRs.
> 
> Indeed, I could. The thing was this was an automated pull request from the Bot. So, I would have to wait the first pull request to be applied, before updating my fork with the Bot Pull request, and then create a second pull request for that part.

You could have cherry picked it into another branch.

**Comment by PoorPocketsMcNewHold** [comment]
> > > You could have split this into two PRs.
> > 
> > 
> > Indeed, I could. The thing was this was an automated pull request from the Bot. So, I would have to wait the first pull request to be applied, before updating my fork with the Bot Pull request, and then create a second pull request for that part.
> 
> You could have cherry picked it into another branch.

That's true. I’m not really that fluid using Github, so I’ve didn’t think of that method. Nonetheless, I’ll wait to an answer from possibly @rugk in case this might not be considered as an issue.

**Comment by rugk** [comment]
Hi @PoorPocketsMcNewHold,
thanks for your first contribution to this project! :tada: :+1: 🏅
I hope you'll like this project and enjoy hacking on it… :smiley:

As others have said, I really would have appreciated it, if you have made two PRs. I'll review it anyway, but this just makes things a little more complicated.

BTW you can (automatically) [let issues close](https://help.github.com/articles/closing-issues-via-commit-messages/) when a PR is merged by adding some "magic" text to your PR _body_. It would be kind, if you could edit your PR to change this.

In case you did not do that already please also subscribe to https://github.com/rugk/mastodon-simplified-federation/issues/36 to get notified for new updates on translations etc. :smiley:

**Comment by PoorPocketsMcNewHold** [comment]
> image changes:
> 
> * do _not_ modify files in `assets`. At least not all (`_optimized`would be okay, of course), as these are partially the source files for actually later editing things. smile
> * as for the other changes, the files are quite little currently anyway, maybe not that optimized, but who cares…
> * al in all, better revert this image edit, that's the easiest thing
> 
> l10n:
> 
> * well… it would still be nice if you could change the indentation. (just replace all "4 spaces" with "2 spaces")

Done. I've noticed the errors explanations messages on the extension translation, would you want them to be translated as well ? (The fr.txt)

File: assets/texts/fr/fr.txt [hunk 1]
```diff
@@ -0,0 +1,189 @@
+{
```
Comment (rugk) [inline assets/texts/fr/fr.txt]: The file name is wrong. It has to be `messages.json`. Then, in a proper text editor (just use something simple like Atom or Notepad++ on Windows, installing takes like 5 minutes), you also get syntax highlighting.

See https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/i18n

File: assets/texts/fr/fr.txt [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  // manifest.json
+  "extensionName": {
+      "message": "Mastodon - Fédération simplifiée !",
+      "description": "Name of the extension.",
+      "hash": "8c716f7803f220414804500b514a1dc0"
+  },
+  "extensionNameShort": {
+      "message": "Mastodon - Fédération simplifiée !",
```
Comment (rugk) [inline assets/texts/fr/fr.txt]: The indentation now is a little broken. You have 6 spaces here, while two spaces on the line above.

Anyway, I get fix that myself, after merging.

**Comment by rugk** [comment]
BTW, I can also suggest you to test the changes locally in your browser, so you'll see how the translation looks/whether it works. (that would have "catched" the `fr.txt` issue)

To test out the code, you need to clone the repo locally (with git submodules) and load it into Firefox. Fortunately, that is quite easy without having to install additional tools. [In the contributing guide](https://github.com/TinyWebEx/common/blob/master/CONTRIBUTING.md#coding), you can see how to do that.

**Comment by rugk** [comment]
@PoorPocketsMcNewHold If you need any pointers/help or have questions, feel free to ask.

**Comment by rugk** [comment]
@PoorPocketsMcNewHold Are you still interested in completing this? If not, that's no problem, just add a notice, so I know what the current status of this PR is.

Also v1.3 is planned for release, so I'd like to include these changes here. See also that there are [some new things to translate](https://github.com/rugk/mastodon-simplified-federation/issues/40#issuecomment-542588829).

**Comment by PoorPocketsMcNewHold** [comment]
Sorry, I’ve been pretty busy those last days. I suppose i’ll will be able to update the translation only later. Surely in the next weeks.

**Comment by rugk** [comment]
Uhm and don't forget [the new things I've linked earlier](https://github.com/rugk/mastodon-simplified-federation/issues/40#issuecomment-542588829).

**Comment by rugk** [comment]
@PoorPocketsMcNewHold Are you still interested in it? If not, that's no problem, just add a notice, so I know what the current status of this issue is.

**Comment by rugk** [comment]
Hmm, this is blocking the [1.3](https://github.com/rugk/mastodon-simplified-federation/milestone/3) release, so I may soon adjust it. @PoorPocketsMcNewHold

**Comment by rugk** [comment]
> The issue however, is that I don't know how to “add” new page to your wiki (And surely shouldn't be supposed to be able to anyway),

There should be a green "New page" button on https://github.com/rugk/mastodon-simplified-federation/wiki. You can just edit the wiki online.

**Comment by rugk** [comment]
Continuing this in https://github.com/rugk/mastodon-simplified-federation/pull/59 now.

**Pull request closed without merging**